src/lib/core/CHIPConfig.h

```diff
@@ -1634,6 +1634,16 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_CONFIG_ICD_OBSERVERS_POOL_SIZE 2
 #endif
 
+/**
+ * @def CHIP_CONFIG_TCP_SUPPORT_ENABLED
+ *  @brief
+ *    Enable (1) or disable (0) support for TCP as a transport protocol for
+ *    operational communications.
+ */
+#ifndef CHIP_CONFIG_TCP_SUPPORT_ENABLED
+#define CHIP_CONFIG_TCP_SUPPORT_ENABLED (INET_CONFIG_ENABLE_TCP_ENDPOINT)
```

Comment:
I think we should only keep one variable if they are always in sync.
 
 If there is no way for TCP_SUPPORT_ENABLED and ENABLE_TCP_ENDPOINT to be different, we should drop one of them. if they can be different, a comment should explain how and why (because as a code reader I would find that surprising)

Reply:
Currently only using INET_CONFIG_ENABLE_TCP_ENDPOINT everywhere.